### PR TITLE
add leaflet-gpx v1.0.0 with git auto-update config

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #6746
@dannyxx001  please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/mpetazzoni/leaflet-gpx
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 120
   - Count of watchers: 26
   - Count of forks: 37
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

cc #6746, cc @mpetazzoni

### CODE REVIEW

**Comment by dannyxx001** [comment]
@PeterDaveHello I think this PR is OK. Please review again, thanks.

File: ajax/libs/leaflet-gpx/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    ]
+  },
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/mpetazzoni/leaflet-gpx.git"
+  },
+  "version": "1.0.0",
+  "description": "GPX layer, targeted at sporting activities by providing access to information such as distance, moving time, pace, elevation, heart rate, etc. ",
+  "author": "mpetazzoni",
+  "license": "BSD 2-clause",
```
Comment (PeterDaveHello) [inline ajax/libs/leaflet-gpx/package.json]: Wrong format there.

Comment (dannyxx001) [inline ajax/libs/leaflet-gpx/package.json]: What do you mean the wrong format?@@

Comment (PeterDaveHello) [inline ajax/libs/leaflet-gpx/package.json]: See `tools/license-list.json` for valid license format.

Comment (dannyxx001) [inline ajax/libs/leaflet-gpx/package.json]: OK, @kennynaoh  I think we should use this `"BSD-2-Clause"`.

**Comment by kennynaoh** [comment]
Thank you both.
@dannyxx001 
Please help me review it again.
Thanks

**Comment by PeterDaveHello** [comment]
@kennynaoh you didn't rebase your branch with the latest master branch.

**Comment by kennynaoh** [comment]
@dannyxx001 
I have rebased my branch to the latest master branch.
Please help me review it again.
Thanks.

**Comment by PeterDaveHello** [comment]
The first line of you commit message isn't too long to include the auto-update info, is it?

**Comment by kennynaoh** [comment]
No, it isn't too long. Thanks.
@dannyxx001 
Please help me review it again.
Thanks.

**Comment by PeterDaveHello** [comment]
No issue reference in the commit message?

**Comment by kennynaoh** [comment]
You mean I need to close the issue?

**Comment by PeterDaveHello** [comment]
@kennynaoh cc ...

**Comment by kennynaoh** [comment]
You mean in the checklist box?
I forgot to shift line, but I updated it.
And I think I have cc author in commit message, right?

**Comment by PeterDaveHello** [comment]
@kennynaoh You should refer the related issue in commit message.

**Comment by kennynaoh** [comment]
Maybe like this:
#6746, cc @...  ?

**Comment by PeterDaveHello** [comment]
@kennynaoh Nope, you can ask @pvnr0082t

**Comment by kennynaoh** [comment]
@PeterDaveHello , @pvnr0082t 
You mean:
```close #6746 , cc @...```?
But I think the issue can not be closed.
Thanks.

**Comment by PeterDaveHello** [comment]
@kennynaoh cc doesn't close issue.

**Comment by PeterDaveHello** [comment]
@kennynaoh https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by kennynaoh** [comment]
Thanks.
I have updated it.
@dannyxx001 
Please help me review this PR again.
Thanks.

File: ajax/libs/leaflet-gpx/1.0.0/gpx.js [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+
+var _MAX_POINT_INTERVAL_MS = 15000;
+var _SECOND_IN_MILLIS = 1000;
+var _MINUTE_IN_MILLIS = 60 * _SECOND_IN_MILLIS;
+var _HOUR_IN_MILLIS = 60 * _MINUTE_IN_MILLIS;
+
+var _DEFAULT_MARKER_OPTS = {
+  startIconUrl: 'pin-icon-start.png',
+  endIconUrl: 'pin-icon-end.png',
+  shadowUrl: 'pin-shadow.png',
```
Comment (PeterDaveHello) [inline ajax/libs/leaflet-gpx/1.0.0/gpx.js]: png files missing.

File: ajax/libs/leaflet-gpx/package.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "leaflet-gpx",
+  "filename": "gpx.min.js",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/mpetazzoni/leaflet-gpx.git",
+    "basePath": "",
+    "files": [
+      "gpx*"
```
Comment (PeterDaveHello) [inline ajax/libs/leaflet-gpx/package.json]: `"*gpx*"` will be more flexible in this case.

Comment (kennynaoh) [inline ajax/libs/leaflet-gpx/package.json]: ```(*gpx|pin-)* ```
Is this correct?
Thanks.

Comment (PeterDaveHello) [inline ajax/libs/leaflet-gpx/package.json]: Fault, don't just simply guess, think about the reason.

Comment (kennynaoh) [inline ajax/libs/leaflet-gpx/package.json]: Because I think whole the ```gpx...``` and ```pin-...``` files should be added.
So I use that form.
Thanks.

**Comment by kennynaoh** [comment]
I change it to ```+(*gpx|pin-)* ```
I think this is correct.
@dannyxx001 
Please help me review this PR again.
Thanks.

**Comment by PeterDaveHello** [comment]
That's bad idea, `pin-*` doesn't look to have its own meaning. You should at least add a png postfix.

**Comment by kennynaoh** [comment]
I think this will be much better.
```"files": [ ```
```"*gpx*", ```
```"*png" ```
```] ```
Is this ok?

**Comment by PeterDaveHello** [comment]
Nope, you added something but also dropped something.

**Comment by kennynaoh** [comment]
How about 
```"files": [ ```
```"*gpx*", ```
```"pin-*png" ```
```] ```

**Comment by kennynaoh** [comment]
or
```"files": [ ```
```"+(*gpx|pin-)*.+(js|map|png)", ```
```] ```

**Comment by PeterDaveHello** [comment]
Just
```js
"files": [
  "*gpx*",
  "pin-*.png"
]
```

**Comment by kennynaoh** [comment]
Thanks.
@dannyxx001 
I have updated it.
Please help me review this PR.
Thanks.

**Comment by kennynaoh** [comment]
ping @dannyxx001

**Comment by dannyxx001** [comment]
@kennynaoh 
Sorry for late review. I think this PR is OK, however, we have to rebase first then we can call Peter.

**Comment by kennynaoh** [comment]
@dannyxx001 
I have rebased to the latest master branch.
Please help me review it again.
Thanks.

**Comment by kennynaoh** [comment]
@dannyxx001 
I have rebased to the latest master branch.
Please help me review it again.
Thanks.

**Comment by dannyxx001** [comment]
@PeterDaveHello I think this PR is OK. Please review again, thanks.

**Pull request merged**